Sources/AppStoreConnectCLI/Files/ResourceFolderManager.swift

```diff
@@ -0,0 +1,26 @@
+// Copyright 2020 Itty Bitty Apps Pty Ltd
+
+import Foundation
+
+protocol ResourceFolderManager {
+    associatedtype T: Codable
+
+    var path: String { get }
+
+    func read() throws -> [T]
+
+    func save(_: [T]) throws
+}
+
+protocol FileNameGettable {
```

Comment:
`Provider` is a little more idiomatic compared to `Gettable`
John Sundell wrote some good thoughts regarding different kinds of protocols here: https://www.swiftbysundell.com/articles/different-categories-of-swift-protocols/